FirebasePerformance/Sources/FPRNanoPbUtils.m

```diff
@@ -0,0 +1,452 @@
+  //  }];
+  //
+  //  return networkMetric;
+  firebase_perf_v1_NetworkRequestMetric networkMetric =
+      firebase_perf_v1_NetworkRequestMetric_init_default;
+  return networkMetric;
+}
+
+firebase_perf_v1_GaugeMetric GetGaugeMetric(NSArray *gaugeData, NSString *sessionId) {
+  //  if (gaugeData == nil || gaugeData.count == 0) {
+  //    return nil;
+  //  }
+  //
+  //  if (sessionId == nil || sessionId.length == 0) {
+  //    return nil;
+  //  }
+  //
+  //  FPRMSGGaugeMetric *gaugeMetric = [FPRMSGGaugeMetric message];
+  //  gaugeMetric.sessionId = sessionId;
+  //  NSMutableArray<FPRMSGCpuMetricReading *> *cpuReadings =
+  //      [[NSMutableArray<FPRMSGCpuMetricReading *> alloc] init];
+  //  NSMutableArray<FPRMSGIosMemoryReading *> *memoryReadings =
+  //      [[NSMutableArray<FPRMSGIosMemoryReading *> alloc] init];
+  //  [gaugeData enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
+  //    if ([obj isKindOfClass:[FPRCPUGaugeData class]]) {
+  //      FPRCPUGaugeData *gaugeData = (FPRCPUGaugeData *)obj;
+  //      FPRMSGCpuMetricReading *cpuReading = [FPRMSGCpuMetricReading message];
+  //      cpuReading.clientTimeUs = gaugeData.collectionTime.timeIntervalSince1970 * USEC_PER_SEC;
+  //      cpuReading.systemTimeUs = gaugeData.systemTime;
+  //      cpuReading.userTimeUs = gaugeData.userTime;
+  //      [cpuReadings addObject:cpuReading];
+  //    }
+  //
+  //    if ([obj isKindOfClass:[FPRMemoryGaugeData class]]) {
+  //      FPRMemoryGaugeData *gaugeData = (FPRMemoryGaugeData *)obj;
+  //      FPRMSGIosMemoryReading *memoryReading = [FPRMSGIosMemoryReading message];
+  //      memoryReading.clientTimeUs = gaugeData.collectionTime.timeIntervalSince1970 *
+  //      USEC_PER_SEC; memoryReading.usedAppHeapMemoryKb =
+  //      (int32_t)BYTES_TO_KB(gaugeData.heapUsed); memoryReading.freeAppHeapMemoryKb =
+  //      (int32_t)BYTES_TO_KB(gaugeData.heapAvailable); [memoryReadings addObject:memoryReading];
+  //    }
+  //  }];
+  //  gaugeMetric.cpuMetricReadingsArray = cpuReadings;
+  //  gaugeMetric.iosMemoryReadingsArray = memoryReadings;
+  //  return gaugeMetric;
+  firebase_perf_v1_GaugeMetric gaugeMetric = firebase_perf_v1_GaugeMetric_init_default;
+  return gaugeMetric;
+}
+
+firebase_perf_v1_ApplicationProcessState ApplicationProcessState(FPRTraceState state) {
+  //  FPRMSGApplicationProcessState processState =
+  //      FPRMSGApplicationProcessState_ApplicationProcessStateUnknown;
+  //  switch (state) {
+  //    case FPRTraceStateForegroundOnly:
+  //      processState = FPRMSGApplicationProcessState_Foreground;
+  //      break;
+  //
+  //    case FPRTraceStateBackgroundOnly:
+  //      processState = FPRMSGApplicationProcessState_Background;
+  //      break;
+  //
+  //    case FPRTraceStateBackgroundAndForeground:
+  //      processState = FPRMSGApplicationProcessState_ForegroundBackground;
+  //      break;
+  //
+  //    default:
+  //      break;
+  //  }
```

Comment:
Should these be commented out? It looks like we were able to bring in ENUMs. Can we respond back with the proper NanoPb structure for this data?

Reply:
Updated.